engine/src/main/scala/cromwell/server/CromwellSystem.scala

```diff
@@ -34,11 +34,15 @@ trait CromwellSystem {
   implicit private final lazy val ec = actorSystem.dispatcher
 
   def shutdownActorSystem(): Future[Terminated] = {
-    Http().shutdownAllConnectionPools() flatMap { _ =>
-      shutdownMaterializerAndActorSystem()
-    } recoverWith {
-      case _ => shutdownMaterializerAndActorSystem()
-    }
+    // If the actor system is already terminated it's already too late for a clean shutdown
+    if (!actorSystem.whenTerminated.isCompleted) {
```

Comment:
ToL: This does seem better that what was there before, but checking the `Future.isCompleted` is false then trying to run another future seems like a race condition. `!isCompleted` doesn't mean that a shutdown hasn't already started.

"ToL" because off the top of my head I don't have anything better at the moment. Someday we should audit all the code and ensure no one is terminating the actor system without going through `def shutdownActorSystem()`, such that this check should _in theory_ never be necessary.

Reply:
Good point, I think I added this because in the event where the root actor crashes at initialization, it escalates to the guardian which itself dies which shuts down the actor system, but we still end up here trying to shut it down again and it can get stuck for some reason.
So in this case it's not even another part of the code it's just a side effect of supervision.